### PR TITLE
Display timestamp in Signer requests details

### PR DESCRIPTION
**Pull request by ngotchac**
Display the request creation date in the Signer Requests details (#2283)

### CODE REVIEW

File: js/src/views/Signer/components/TransactionSecondaryDetails/TransactionSecondaryDetails.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component, PropTypes } from 'react';
+
+import ReactTooltip from 'react-tooltip';
+import DescriptionIcon from 'material-ui/svg-icons/action/description';
+import GasIcon from 'material-ui/svg-icons/maps/local-gas-station';
```
Comment (derhuerst) [inline js/src/views/Signer/components/TransactionSecondaryDetails/TransactionSecondaryDetails.js]: :stuck_out_tongue_closed_eyes:

File: js/src/views/Signer/components/TransactionPending/TransactionPending.js [hunk 2]
```diff
@@ -65,20 +65,24 @@ export default class TransactionPending extends Component {
...
             className={ styles.transactionDetails }
             totalValue={ totalValue }>
-            <div className={ styles.iconsContainer }>
-              { this.renderGasPrice() }
-              { this.renderData() }
-            </div>
-            <div className={ styles.expandedContainer }>
-              { this.renderDataExpanded() }
-            </div>
+            <TransactionSecondaryDetails
```
Comment (derhuerst) [inline js/src/views/Signer/components/TransactionPending/TransactionPending.js]: Is it really worth creating an etirely new component? I don't have strong opinions on this, but it feels bloated.

File: js/src/views/Signer/components/TransactionSecondaryDetails/TransactionSecondaryDetails.css [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+.data.noData {
+  cursor: text;
+}
+
+
+.dataTooltip {
+  word-wrap: break-word;
+  max-width: 400px;
```
Comment (derhuerst) [inline js/src/views/Signer/components/TransactionSecondaryDetails/TransactionSecondaryDetails.css]: Using `em` as a unit for layout-related things will help increate modularity & accessibility.

File: js/src/views/Signer/reducers/requests.js [hunk 4]
```diff
@@ -108,3 +108,8 @@ function setIsSending (pending, id, isSending) {
     return p;
   }).slice();
 }
+
+function addTimestamp (request) {
+  request.date = new Date();
```
Comment (derhuerst) [inline js/src/views/Signer/reducers/requests.js]: This modifies the `request` object and can therefore later have unintended side effects. I'd propose `(r) => {request: r, timestamp: new Date()}`. If you don't need the request, `(r) => new Date()`.

File: js/src/views/Signer/components/TransactionSecondaryDetails/TransactionSecondaryDetails.css (hunk 3 above)
Comment (jacogr) [inline js/src/views/Signer/components/TransactionSecondaryDetails/TransactionSecondaryDetails.css]: Agreed. However, this Signer is a nightmare as the layouts are, we need a separate issue for re-working the layouts. As-is this is the best approach to take here unless you want to break everything in the signer display.

Comment (derhuerst) [inline js/src/views/Signer/components/TransactionSecondaryDetails/TransactionSecondaryDetails.css]: fair enough

File: js/src/views/Signer/components/TransactionPending/TransactionPending.js (hunk 2 above)
Comment (jacogr) [inline js/src/views/Signer/components/TransactionPending/TransactionPending.js]: As it stands, for the Signer is is probably the best approach. Everything is very, very, very modular and re-used among the different Pending/Completed layouts. Best way of fitting in with the style that is there.

File: js/src/views/Signer/reducers/requests.js (hunk 4 above)
Comment (ngotchac) [inline js/src/views/Signer/reducers/requests.js]: The thing is that this new format would imply many changes for Components using theses requests. Maybe add this `timestamp` field at a lower level, like `request.payload.timestamp` ?

**Comment by jacogr** [comment]
@ngotchac Can I add and issue to fix here while you are digging around - it would seem like the Signer stuff doesn't pop up in most-recent first on-top. It should be a simple fix, not sure why it goes awry on my side, I was sure it actually had the most recent on-top.

https://github.com/ethcore/parity/issues/2343

**Pull request merged**